python/hidet/ir/primitives/cpu/math/bfloat16.py

```diff
@@ -0,0 +1,98 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from hidet.ir.expr import Expr, Call
+from hidet.ir.type import FuncType
+from hidet.ir.primitives.func import register_primitive_function, primitive_func_pool
+from hidet.ir.primitives.math import MathFunctionSet, register_math_function_set
+
+
+class CPUBFloat16MathFunctionSet(MathFunctionSet):
+    # pylint: disable=abstract-method
+    def register(self):
+        entries = {
+            'sin': ['sin', 1],
+            'cos': ['cos', 1],
+            'exp': ['exp', 1],
+            'sqrt': ['sqrt', 1],
+            'rsqrt': ['rsqrt', 1],
+            'log': ['log', 1],
+            'round': ['round', 1],
+            'ceil': ['ceil', 1],
+            'floor': ['floor', 1],
+            'tanh': ['tanh', 1],
+            'erf': ['erf', 1],
+            'min': ['min', 2],
+            'max': ['max', 2],
+            'pow': ['pow', 2],
+            'fma': ['fma', 3],
+        }
+
+        for name, (codegen_name, num_args) in entries.items():
+            register_primitive_function(
+                name='cpu_bf16_{}'.format(name),
+                codegen_name=codegen_name,
+                func_or_type=FuncType(param_types=['bfloat16'] * num_args, ret_type='bfloat16'),
+            )
+
+    def call(self, name: str, *args) -> Expr:
+        entry = primitive_func_pool.lookup_by_name(name)
+        return Call(entry.var, args)
+
+    def sin(self, a: Expr) -> Expr:
+        return self.call('cpu_bf16_sin', a)
```

Comment:
Might be better to make the name conversion consistent with the float16?

```suggestion
        return self.call('sin', a)
```
and update the `self.call(...)`